// restaurant-package/index.js
var orders = [
  {
    code : 1,
    client : 'Pippo',
    list : '2 Pizze margherita, 1 Ferrarelle, 1 Vino rosso',
    bill : 25.00,
    status : 'closed'
  },
  {
    code : 2,
    client : 'Caio',
    list : '1 Antipasto,1 Penne al sugo, 1 Spaghetti alla carbonara,1 Ferrarelle, 1 Vino rosso, 2 caffè',
    bill : 45.00,
    status : 'ready'
  },
  {
    code : 3,
    client : 'Sempronio',
    list : '3 Fritto misto mare,1 Pesce spada, 2 Aragosta,3 Ostrica, 2 Vino bianco, 3 caffè',
    bill : 77.00,
    status : 'new'
  },
];


this.getOrderIndex = function (code){
  if (isNaN(code) || code < 0){
    return "error";
  } else {
      for (var index  in orders){
        var order = orders[index];
        if(order.code == code){
          return index;
        }
      }
    }
  return null;
}

exports.ordersList = function (){
  return orders;
}

exports.newOrder = function (object){
  orders.push(
    {
      code : orders[orders.length-1].code + 1,
      client : object.client,
      list : object.list,
      bill : object.bill,
      status : 'new'
    }
  );
  return null;
}

exports.deleteOrder = function (code){
  var order = this.getOrderIndex(code);
  orders.splice(order,1);
  return null;
}

exports.getOrderByStatus = function (orderStatus){
  var orderByStatus = [];

  for (var index  in orders){
    var order = orders[index];
    if(order.status == orderStatus){
      orderByStatus.push(order);
    }
  }
  return orderByStatus;
}

exports.setOrder = function (code,setting){
  var order = orders[this.getOrderIndex(code)];
  order.status = setting;
  return null;
}

exports.profit = function (){
  var profit = 0;
  for (var index in orders){
    profit += orders[index].bill;
  }
  return profit;
}

exports.ordersByClients = function(client){
  var ordersByClients = [];
  
   for (var index in orders){
     var user = orders[index];
     if(user.client == client){
       ordersByClients.push(user);
     }
   }
    return ordersByClients;
}
